src/Simulation/Simulators/ToffoliSimulator/ToffoliSimulator.cs

```diff
@@ -20,478 +17,29 @@ public enum ToffoliDumpFormat
-            {
-                equivalentToX = false;
-                safeToControl = true;
-            }
-            // If power > 1, then we're an odd multiple of pi/4, so illegal
-            else if (power > 1)
-            {
-                throw new InvalidOperationException($"The Toffoli simulator can only perform rotations of multiples of pi/2.");
-            }
-            // If power == 1, then we're an odd multiple of pi/2, so X if the axis is X, 
-            // I if the axis is I, and illegal otherwise.
-            else if (power == 1)
-            {
-                switch (axis)
-                {
-                    case Pauli.PauliI:
-                        equivalentToX = false;
-                        safeToControl = false;
-                        break;
-                    case Pauli.PauliX:
-                        equivalentToX = true;
-                        safeToControl = false;
-                        break;
-                    case Pauli.PauliY:
-                    case Pauli.PauliZ:
-                        throw new InvalidOperationException($"The Toffoli simulator can only perform non-identity rotations around the X axis.");
-                }
-            }
-            // If power == 0, then we're a multiple of pi, possible an even multiple.
-            else
-            {
-                equivalentToX = false;
-                safeToControl = (numerator % 2 == 0);
-            }
-
-            return (equivalentToX, safeToControl);
-        }
-
-        /// <summary>
-        /// Mark a qubit as being borrowed, tracking its state before borrowing.
-        /// </summary>
-        /// <param name="q">The qubit being borrowed.</param>
-        internal void BorrowQubit(Qubit q)
-        {
-            Stack<bool> stack;
-            if (!borrowedQubitStates.TryGetValue(q.Id, out stack))
-            {
-                stack = new Stack<bool>();
-                borrowedQubitStates.Add(q.Id, stack);
-            }
-            stack.Push(State[q.Id]);
-        }
-
-        /// <summary>
-        /// Mark a group of qubits as being borrowed, tracking their states before borrowing.
-        /// </summary>
-        /// <param name="qs">The qubits being borrowed.</param>
-        internal void BorrowQubits(IEnumerable<Qubit> qs)
-        {
-            foreach (var q in qs)
-            {
-                BorrowQubit(q);
-            }
-        }
-
-        /// <summary>
-        /// Validate that a borrowed qubit is being returned in its original state.
-        /// </summary>
-        /// <param name="q">The qubit being returned.</param>
-        internal void ReturnQubit(Qubit q)
-        {
-            var stack = borrowedQubitStates[q.Id];
-            if (stack.Pop() != State[q.Id])
-            {
-                throw new ExecutionFailException("Borrowed qubit returned not in the original state");
-            }
+    public class ToffoliSimulator : QuantumProcessorDispatcher {
```

Comment:
>{ [](start = 63, length = 1)

From other *.cs files in this repo it seems that the coding style for C# is to put the opening brace on the new line.

Reply:
Thanks for pointing this out. I will fix this in the next commit.